lib/index.js

```diff
@@ -327,14 +331,20 @@ api.getPublicKeys = function(id, actor, options, callback) {
 // FIXME: sync private key label to public key label
 api.updatePublicKey = function(actor, publicKey, callback) {
   async.auto({
-    checkPermission: function(callback) {
+    clean: function(callback) {
+      // Removing owner forces the get function to be called in checkPermission,
+      // so the owner checked is the one in the db, not the key passed in.
+      delete publicKey.owner;
```

Comment:
On further reflection and conversation with @mattcollier, I don't think we should do this change. The caller of the API needs to know not to send in `owner` rather than having us delete it here. @mattcollier can share more thoughts if he wants.